diff --git theme/calendar-month-node.tpl.php theme/calendar-month-node.tpl.php
index 6ffcb90..0b4692e 100644
--- theme/calendar-month-node.tpl.php
+++ theme/calendar-month-node.tpl.php
@@ -39,7 +39,7 @@
  */
 ?>
 <div class="view-item view-item-<?php print $view->name ?>">
-  <div class="calendar monthview" id="<?php print $node->date_id ?>">
+  <div class="<?php print $class; ?>" id="<?php print $node->date_id ?>">
     <?php print theme('calendar_stripe_stripe', $node); ?>
     <?php foreach ($fields as $field): ?>
       <div id="<?php print $field['id']; ?>" class="view-field view-data-<?php print $field['id'] ?>">
diff --git theme/theme.inc theme/theme.inc
index 19e01a3..0548dbc 100644
--- theme/theme.inc
+++ theme/theme.inc
@@ -577,6 +577,7 @@ function template_preprocess_calendar_node(&$vars) {
  * Format an calendar month node for display.
  */
 function template_preprocess_calendar_month_node(&$vars) {
+  $vars['class'] = 'calendar monthview';
   template_preprocess_calendar_node($vars);
 }
 
