diff --git theme/calendar-month-node.tpl.php theme/calendar-month-node.tpl.php
index 0b4692e..6879fff 100644
--- theme/calendar-month-node.tpl.php
+++ theme/calendar-month-node.tpl.php
@@ -38,7 +38,7 @@
  * @see template_preprocess_calendar_month_node.
  */
 ?>
-<div class="view-item view-item-<?php print $view->name ?>">
+<div class="view-item view-item-<?php print $view->name ?> view-item-<?php print $zebra; ?>">
   <div class="<?php print $class; ?>" id="<?php print $node->date_id ?>">
     <?php print theme('calendar_stripe_stripe', $node); ?>
     <?php foreach ($fields as $field): ?>
