diff --git a/includes/calendar_plugin_display_attachment.inc b/includes/calendar_plugin_display_attachment.inc
index 40567fd..47c8e3e 100644
--- a/includes/calendar_plugin_display_attachment.inc
+++ b/includes/calendar_plugin_display_attachment.inc
@@ -155,17 +155,6 @@ class calendar_plugin_display_attachment extends views_plugin_display_attachment
     }
     parent::query();    
   }
-  
-  /**
-   * Set default values for the display.
-   */
-  function options(&$display) {
-    parent::options($display);
-    $display->display_options['inherit_argments'] = TRUE;
-    $display->display_options['inherit_filters'] = TRUE;
-    $display->display_options['attachment_position'] = 'after';
-    $display->display_options['calendar_type'] = $this->calendar_type();
-  }  
 
   /**
    * Add custom option definitions.
@@ -173,6 +162,9 @@ class calendar_plugin_display_attachment extends views_plugin_display_attachment
   function option_definition () {
     $options = parent::option_definition();
     $options['calendar_type'] = array('default' => $this->calendar_type());
+    $options['inherit_argments']['default'] = TRUE;
+    $options['inherit_filters']['default'] = TRUE;
+    $options['attachment_position']['default'] = 'after';
     return $options;
   }
   
diff --git a/includes/calendar_plugin_display_block.inc b/includes/calendar_plugin_display_block.inc
index 79df6a1..9b9d00e 100644
--- a/includes/calendar_plugin_display_block.inc
+++ b/includes/calendar_plugin_display_block.inc
@@ -44,24 +44,18 @@ class calendar_plugin_display_block extends views_plugin_display_block {
     }
     return parent::defaultable_sections($section);
   }
-
-  function options(&$display) {
-    parent::options($display);
-    $display->display_options['displays'] = array();
-    $display->display_options['style_plugin'] = 'calendar_nav';
-    $display->display_options['items_per_page'] = 0;
-    $display->display_options['row_plugin'] = '';
-    $display->display_options['defaults']['style_plugin'] = FALSE;
-    $display->display_options['defaults']['style_options'] = FALSE;
-    $display->display_options['defaults']['items_per_page'] = FALSE;
-    $display->display_options['defaults']['row_plugin'] = FALSE;
-    $display->display_options['defaults']['row_options'] = FALSE;
-  } 
   
   function option_definition () {
     $options = parent::option_definition();
 
     $options['style_plugin'] = array('default' => 'calendar_nav');
+    $options['items_per_page']['default'] = 0;
+    $options['row_plugin']['default'] = '';
+    $options['defaults']['default']['style_plugin'] = FALSE;
+    $options['defaults']['default']['style_options'] = FALSE;
+    $options['defaults']['default']['items_per_page'] = FALSE;
+    $options['defaults']['default']['row_plugin'] = FALSE;
+    $options['defaults']['default']['row_options'] = FALSE;
     return $options;
   }
 
diff --git a/includes/calendar_plugin_display_page.inc b/includes/calendar_plugin_display_page.inc
index 0ea57bc..e2f8117 100644
--- a/includes/calendar_plugin_display_page.inc
+++ b/includes/calendar_plugin_display_page.inc
@@ -41,19 +41,6 @@ class calendar_plugin_display_page extends views_plugin_display_page {
     }
     return parent::defaultable_sections($section);
   }
-
-  function options(&$display) {
-    parent::options($display);
-    $display->display_options['displays'] = array();
-    $display->display_options['style_plugin'] = 'calendar_nav';
-    $display->display_options['items_per_page'] = 0;
-    $display->display_options['row_plugin'] = '';
-    $display->display_options['defaults']['style_plugin'] = FALSE;
-    $display->display_options['defaults']['style_options'] = FALSE;
-    $display->display_options['defaults']['row_plugin'] = FALSE;
-    $display->display_options['defaults']['row_options'] = FALSE;
-    $display->display_options['defaults']['items_per_page'] = FALSE;
-  } 
   
   function option_definition () {
     $options = parent::option_definition();
@@ -63,6 +50,12 @@ class calendar_plugin_display_page extends views_plugin_display_page {
     $options['calendar_popup'] = array('default' => 0);
     $options['calendar_date_link'] = array('default' => '');
     $options['style_plugin'] = array('default' => 'calendar_nav');
+    $options['items_per_page']['default'] = 0;
+    $options['row_plugin']['default'] = '';
+    $options['defaults']['default']['style_options'] = FALSE;
+    $options['defaults']['default']['row_plugin'] = FALSE;
+    $options['defaults']['default']['row_options'] = FALSE;
+    $options['defaults']['default']['items_per_page'] = FALSE;
     return $options;
   }
 
