Index: contributions/modules/date/date/date_handler_field_multiple.inc
--- contributions/modules/date/date/date_handler_field_multiple.inc Base (1.1.2.12)
+++ contributions/modules/date/date/date_handler_field_multiple.inc Locally Modified (Based On 1.1.2.12)
@@ -10,13 +10,25 @@
  */
 class date_handler_field_multiple extends content_handler_field_multiple {
   
-  function options(&$options) {
-    parent::options($options);
+  function option_definition() {
+    $options = parent::option_definition();
     $options['repeat'] = array(
-      'show_repeat_rule' => '',
+      'contains' => array(
+        'show_repeat_rule' => array('default' => ''),
+      )
       );
-    $options['multiple']['multiple_to'] = ''; 
-    $options['fromto']['fromto'] = 'both'; 
+    $options['multiple'] = array(
+      'contains' => array(
+        'multiple_to' => array('default' => ''),
+      )
+    );
+    $options['fromto'] = array(
+      'contains' => array(
+        'fromto' => array('default' => 'both'),
+      )
+    );
+
+    return $options;
   }
   
   /**
Index: contributions/modules/date/includes/date_api_argument_handler.inc
--- contributions/modules/date/includes/date_api_argument_handler.inc Base (1.1.2.38)
+++ contributions/modules/date/includes/date_api_argument_handler.inc Locally Modified (Based On 1.1.2.38)
@@ -42,12 +42,14 @@
   /**
    * Default value for the date_fields option.
    */
-  function options(&$options) {
-    parent::options($options);
-    $options['date_fields'] = array();
-    $options['year_range'] = '-3:+3';
-    $options['date_method'] = 'OR';
-    $options['granularity'] = 'month';
+  function option_definition() {
+    $options = parent::option_definition();
+    $options['date_fields'] = array('default' => '');
+    $options['year_range'] = array('default' => '-3:+3');
+    $options['date_method'] = array('default' => 'OR');
+    $options['granularity'] = array('default' => 'month');
+
+    return $options;
   }
 
   /**
Index: contributions/modules/date/includes/date_api_filter_handler.inc
--- contributions/modules/date/includes/date_api_filter_handler.inc Base (1.1.2.41)
+++ contributions/modules/date/includes/date_api_filter_handler.inc Locally Modified (Based On 1.1.2.41)
@@ -53,19 +53,15 @@
   }
 
   // Set default values for the date filter.
-  function options(&$options) {
-    parent::options($options);
-    $options['date_fields'] = array();
-    $options['date_method'] = 'OR';
-    $options['granularity'] = 'day';
-    $options['form_type'] = 'date_select';
-    $options['default_date'] = '';
-    $options['default_to_date'] = '';
-    $options['year_range'] = '-3:+3';    
-  }
-  
   function option_definition() {
     $options = parent::option_definition();
+    $options['date_fields'] = array('default' => '');
+    $options['date_method'] = array('default' => 'OR');
+    $options['granularity'] = array('default' => 'day');
+    $options['form_type'] = array('default' => 'date_select');
+    $options['default_date'] = array('default' => '');
+    $options['default_to_date'] = array('default' => '');
+    $options['year_range'] = array('default' => '-3:+3');
     return $options;
   }
 
Index: contributions/modules/date/includes/date_plugin_display_attachment.inc
--- contributions/modules/date/includes/date_plugin_display_attachment.inc Base (1.1.2.2)
+++ contributions/modules/date/includes/date_plugin_display_attachment.inc Locally Modified (Based On 1.1.2.2)
@@ -19,16 +19,18 @@
     return parent::defaultable_sections($section);
   }
 
-  function options(&$display) {
-    parent::options($display);
-    $display->display_options['style_plugin'] = 'date_nav';
-    $display->display_options['items_per_page'] = 0;
-    $display->display_options['row_plugin'] = '';
-    //$display->display_options['defaults']['style_plugin'] = FALSE;
-    $display->display_options['defaults']['style_options'] = FALSE;
-    $display->display_options['defaults']['items_per_page'] = FALSE;
-    $display->display_options['defaults']['row_plugin'] = FALSE;
-    $display->display_options['defaults']['row_options'] = FALSE;
+  function option_definition() {
+    $options = parent::option_definition();
+    $options['style_plugin']['default'] = 'date_nav';
+    $options['items_per_page']['default'] = 0;
+    $options['row_plugin']['default'] = '';
+    //$options['defaults']['default']['style_plugin'] = FALSE;
+    $options['defaults']['default']['style_options'] = FALSE;
+    $options['defaults']['default']['items_per_page'] = FALSE;
+    $options['defaults']['default']['row_plugin'] = FALSE;
+    $options['defaults']['default']['row_options'] = FALSE;
+
+    return $options;
   } 
 }
 
